### PR TITLE
Allow user to add new table

### LINKED ISSUE #109: Feature: Allow user to add new table
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

- Click "Add Table" button on `Navbar`,
- Get title from user via dialog box,
- Create logic on Web Frontend and Web Backend to process given data,
- Display new table on Web Frontend automatically,

**Describe alternatives you've considered**

**Additional context**
End2End

### PR DESCRIPTION
**Pull request by Lukash88**
## Summary

Add creating new table feature.

## Control List

- [x] Code formatting checked,
- [x] Coding Guidelines compliance checked,
- [ ] Documentation updated (if necessary),
- [ ] Automated tests updated (if necessary),

## Screenshots

image

### CODE REVIEW

File: binder-web-backend/Binder.Api/Controllers/TablesController.cs [hunk 1]
```diff
@@ -27,5 +27,11 @@ public ActionResult<DefaultTable[]> Get()
         {
             return Ok(_service.GetAllTables());
         }
+
+        [HttpPost]
+        public ActionResult<DefaultTable> Create(string tableName)
```
Comment (StraykerPL) [inline binder-web-backend/Binder.Api/Controllers/TablesController.cs]: Stick to naming convention of the file - if methods have requests names, than use request name here as well.

File: binder-web-backend/Binder.Application/Models/Interfaces/IDefaultTableService.cs [hunk 2]
```diff
@@ -7,5 +7,7 @@ public interface IDefaultTableService
         DefaultTable GetTable(int tableId);
 
         ICollection<DefaultTable> GetAllTables();
+
+        DefaultTable Create(string tableName);
```
Comment (StraykerPL) [inline binder-web-backend/Binder.Application/Models/Interfaces/IDefaultTableService.cs]: Same, naming convention.

File: binder-web-frontend/src/assets/i18n/en.json [hunk 3]
```diff
@@ -17,5 +21,10 @@
     "domainButtons": {
         "options": "Options",
         "about": "About"
+    },
+    "addTableDialog": {
+        "title": "Add Table",
+        "content": "Enter name of new table:",
+        "add": "Add"
```
Comment (StraykerPL) [inline binder-web-frontend/src/assets/i18n/en.json]: You can move this to general section.

File: binder-web-frontend/src/pages/home/components/navbar/navbar.component.ts [hunk 4]
```diff
@@ -1,13 +1,56 @@
...
+  tableName: string = '';
   showHideColumnButtonVisibility: boolean = false;
   resetViewButtonVisibility: boolean = false;
-  
-  constructor() {}
+
+  constructor(private dialog: MatDialog, private tableService: TablesService) { }
+
+  openDialog(): void {
+    let dialogRef = this.dialog.open(TableDialogComponent, {
```
Comment (StraykerPL) [inline binder-web-frontend/src/pages/home/components/navbar/navbar.component.ts]: Make it const.

Comment (StraykerPL) [inline binder-web-frontend/src/pages/home/components/navbar/navbar.component.ts]: Move this settings to some providable settings object in external file.

File: binder-web-frontend/src/pages/home/components/navbar/navbar.component.ts [hunk 5]
```diff
@@ -1,13 +1,56 @@
...
+      width: '18rem',
+      height: '16rem',
+      data: { name: this.tableName },
+      position: { top: '30vh', left:'50vw' }
+    });
+
+    dialogRef.afterClosed().subscribe((result) => {
```
Comment (StraykerPL) [inline binder-web-frontend/src/pages/home/components/navbar/navbar.component.ts]: Make proper subscription handling, like in addTable method.

File: binder-web-frontend/src/pages/home/components/table-dialog/table-dialog.component.scss [hunk 6]
```diff
@@ -0,0 +1,23 @@
+@use "variables";
+
+.dialog {
+    background-color: variables.$color-raisin-black;
+    box-shadow: 0px 0px 16px #00000025;
```
Comment (StraykerPL) [inline binder-web-frontend/src/pages/home/components/table-dialog/table-dialog.component.scss]: This Hex value has a little too many numbers, we don't need such binary precision.

File: binder-web-frontend/src/pages/home/components/table-dialog/table-dialog.component.scss [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+
+    &__buttons {
+        &-btn {
+            user-select: none;
+            width: 5rem;
+            height: 1.6rem;
+            margin-left: 1rem;
+            border-radius: 0.8rem;
```
Comment (StraykerPL) [inline binder-web-frontend/src/pages/home/components/table-dialog/table-dialog.component.scss]: That's cool, but for precision you can make it PX.

File: binder-web-frontend/src/pages/home/components/table-dialog/table-dialog.component.scss [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+.dialog {
+    background-color: variables.$color-raisin-black;
+    box-shadow: 0px 0px 16px #00000025;
+
+    &__buttons {
+        &-btn {
+            user-select: none;
+            width: 5rem;
+            height: 1.6rem;
+            margin-left: 1rem;
```
Comment (StraykerPL) [inline binder-web-frontend/src/pages/home/components/table-dialog/table-dialog.component.scss]: Is it necessary to use margin? Can it be replaced with padding?

File: binder-web-frontend/src/pages/home/components/table-dialog/table-dialog.component.ts [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+
+@Component({
+  selector: 'app-table-dialog',
+  templateUrl: './table-dialog.component.html',
+  styleUrls: ['./table-dialog.component.scss']
+})
+export class TableDialogComponent {
+
+  constructor( 
+    public dialogRef: MatDialogRef<TableDialogComponent>,
```
Comment (StraykerPL) [inline binder-web-frontend/src/pages/home/components/table-dialog/table-dialog.component.ts]: Is it possible to make this values private? Will dialog box still work after changes?

File: binder-web-frontend/src/pages/pages.module.ts [hunk 10]
```diff
@@ -35,8 +46,10 @@ import { MatIconModule } from '@angular/material/icon';
     DomainButtonsComponent,
     HeaderComponent,
     SidePanelComponent,
-    MainSpaceComponent
+    MainSpaceComponent,
+    TableDialogComponent,
   ],
+  entryComponents: [TableDialogComponent],
```
Comment (StraykerPL) [inline binder-web-frontend/src/pages/pages.module.ts]: This feature is deprecated, please remove.

**Review by StraykerPL — CHANGES_REQUESTED** [review]

File: binder-web-backend/Binder.Api/Controllers/TablesController.cs (hunk 1 above)
Comment (Lukash88) [reply binder-web-backend/Binder.Api/Controllers/TablesController.cs]: Resolved 09113b3.

File: binder-web-frontend/src/assets/i18n/en.json (hunk 3 above)
Comment (Lukash88) [reply binder-web-frontend/src/assets/i18n/en.json]: Resolved 09113b3.

File: binder-web-frontend/src/pages/home/components/navbar/navbar.component.ts (hunk 4 above)
Comment (Lukash88) [reply binder-web-frontend/src/pages/home/components/navbar/navbar.component.ts]: Resolved 09113b3.

Comment (Lukash88) [reply binder-web-frontend/src/pages/home/components/navbar/navbar.component.ts]: Resolved 09113b3.

File: binder-web-frontend/src/pages/home/components/navbar/navbar.component.ts (hunk 5 above)
Comment (Lukash88) [reply binder-web-frontend/src/pages/home/components/navbar/navbar.component.ts]: Resolved 09113b3.

File: binder-web-frontend/src/pages/home/components/table-dialog/table-dialog.component.scss (hunk 6 above)
Comment (Lukash88) [reply binder-web-frontend/src/pages/home/components/table-dialog/table-dialog.component.scss]: Resolved 09113b3.

File: binder-web-frontend/src/pages/home/components/table-dialog/table-dialog.component.scss (hunk 7 above)
Comment (Lukash88) [reply binder-web-frontend/src/pages/home/components/table-dialog/table-dialog.component.scss]: Resolved 09113b3.

File: binder-web-frontend/src/pages/home/components/table-dialog/table-dialog.component.scss (hunk 8 above)
Comment (Lukash88) [reply binder-web-frontend/src/pages/home/components/table-dialog/table-dialog.component.scss]: Resolved 09113b3.

File: binder-web-frontend/src/pages/home/components/table-dialog/table-dialog.component.ts (hunk 9 above)
Comment (Lukash88) [reply binder-web-frontend/src/pages/home/components/table-dialog/table-dialog.component.ts]: Resolved 09113b3.

File: binder-web-frontend/src/pages/pages.module.ts (hunk 10 above)
Comment (Lukash88) [reply binder-web-frontend/src/pages/pages.module.ts]: Resolved 09113b3.

File: binder-web-backend/Binder.Application/Models/Interfaces/IDefaultTableService.cs (hunk 2 above)
Comment (Lukash88) [reply binder-web-backend/Binder.Application/Models/Interfaces/IDefaultTableService.cs]: Resolved 09113b3.

File: binder-web-frontend/src/assets/i18n/en.json (hunk 3 above)
Comment (StraykerPL) [reply binder-web-frontend/src/assets/i18n/en.json]: I meant only the commented line :)

File: binder-web-frontend/src/pages/home/components/navbar/navbar.component.ts (hunk 5 above)
Comment (StraykerPL) [reply binder-web-frontend/src/pages/home/components/navbar/navbar.component.ts]: Missing unsubscribe handling.

File: binder-web-frontend/src/pages/home/components/table-dialog/table-dialog.component.scss (hunk 7 above)
Comment (StraykerPL) [reply binder-web-frontend/src/pages/home/components/table-dialog/table-dialog.component.scss]: Again, I meant only border-radius line ;)

File: binder-web-frontend/src/pages/home/components/table-dialog/table-dialog.component.ts (hunk 9 above)
Comment (StraykerPL) [reply binder-web-frontend/src/pages/home/components/table-dialog/table-dialog.component.ts]: I meant both values in constructor, that's why I used plural "values" word ;)

File: binder-web-frontend/src/app/app.module.ts [hunk 11]
```diff
@@ -23,6 +23,7 @@ import { TranslocoRootModule } from './transloco-root.module';
     ApiModule,
     HttpClientModule,
     TranslocoRootModule,
+    FormsModule,
```
Comment (StraykerPL) [inline binder-web-frontend/src/app/app.module.ts]: Double importing of FormsModule, in app.module and pages.module. Is it intended?

File: binder-web-frontend/src/pages/home/components/navbar/navbar.component.ts [hunk 12]
```diff
@@ -1,13 +1,58 @@
 import { Component } from '@angular/core';
+import { MatDialog } from '@angular/material/dialog';
+import { Subject, takeUntil } from 'rxjs';
+import { TableDialogComponent } from '../table-dialog/table-dialog.component';
```
Comment (StraykerPL) [inline binder-web-frontend/src/pages/home/components/navbar/navbar.component.ts]: Make this import absolute-pathed.

File: binder-web-frontend/src/shared/consts/appConsts.ts [hunk 13]
```diff
@@ -0,0 +1,6 @@
+export const dialogConfig = {
+  width: '17rem',
+  height: '16rem',
+  data: { name: '' },
+  position: { top: '30vh', left: '50vw' },
```
Comment (StraykerPL) [inline binder-web-frontend/src/shared/consts/appConsts.ts]: It appeared, that Angular Material will center dialogbox automatically without position value in config - so you can remove it.

**Review by StraykerPL — CHANGES_REQUESTED** [review]

File: binder-web-frontend/src/pages/home/components/table-dialog/table-dialog.component.ts [hunk 14]
```diff
@@ -0,0 +1,18 @@
+import { Component, Inject } from '@angular/core';
+import { MAT_DIALOG_DATA,  MatDialogRef } from '@angular/material/dialog';
+
+@Component({
+  selector: 'app-table-dialog',
```
Comment (StraykerPL) [inline binder-web-frontend/src/pages/home/components/table-dialog/table-dialog.component.ts]: Remove selector's prefix.

File: binder-web-frontend/src/assets/i18n/en.json (hunk 3 above)
Comment (Lukash88) [reply binder-web-frontend/src/assets/i18n/en.json]: Resolved 956ff6a6a.

File: binder-web-frontend/src/pages/home/components/table-dialog/table-dialog.component.scss (hunk 7 above)
Comment (Lukash88) [reply binder-web-frontend/src/pages/home/components/table-dialog/table-dialog.component.scss]: Resolved 956ff6a6a.

File: binder-web-frontend/src/pages/home/components/table-dialog/table-dialog.component.ts (hunk 9 above)
Comment (Lukash88) [reply binder-web-frontend/src/pages/home/components/table-dialog/table-dialog.component.ts]: Resolved 956ff6a6a.

File: binder-web-frontend/src/app/app.module.ts (hunk 11 above)
Comment (Lukash88) [reply binder-web-frontend/src/app/app.module.ts]: Resolved 956ff6a6a.

File: binder-web-frontend/src/pages/home/components/navbar/navbar.component.ts (hunk 12 above)
Comment (Lukash88) [reply binder-web-frontend/src/pages/home/components/navbar/navbar.component.ts]: Resolved 956ff6a6a.

File: binder-web-frontend/src/shared/consts/appConsts.ts (hunk 13 above)
Comment (Lukash88) [reply binder-web-frontend/src/shared/consts/appConsts.ts]: Resolved 956ff6a6a.

File: binder-web-frontend/src/pages/home/components/table-dialog/table-dialog.component.ts (hunk 14 above)
Comment (Lukash88) [reply binder-web-frontend/src/pages/home/components/table-dialog/table-dialog.component.ts]: Resolved 956ff6a6a.

File: binder-web-frontend/src/pages/home/components/navbar/navbar.component.ts (hunk 5 above)
Comment (Lukash88) [reply binder-web-frontend/src/pages/home/components/navbar/navbar.component.ts]: Resolved 956ff6a6a.

**Review by StraykerPL — APPROVED** [review]

**Pull request merged**